### ISSUE TITLE
Increase test coverage of test_tcp_internals.py on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8584]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8584|
|Type|enhancement|
|Created|2016-07-10 05:48:10Z|
|Branch|https://github.com/twisted/twisted/tree/8584-rodrigc-tcp|

test_tcp_internals.py should be enabled on Python 3

<details><summary>Searchable metadata</summary>

```
trac-id__8584 8584
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8584_rodrigc_tcp 8584-rodrigc-tcp
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468129690938535 1468129690938535
changetime__1468259369629439 1468259369629439
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/313

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

Changes look good.

I have sent the branch again to buildbot. Hope this time all tests will pass and coverage will be correctly reported.

Please merge if all is ok.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @adiroiban|
|-|-|

codecov is still blocking the merge

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

The changes are failing on windows

https://buildbot.twistedmatrix.com/builders/windows7-64-py2.7-coverage/builds/775

this needs to be fixed :)

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

The Windows tests are working but codecov is still blocking the commit.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @rodrigc|
|-|-|

win2012r2-64-py2.7-coverage has failed to start and windows7-64-py2.7-coverage is broken ... so windows coverage was not reported.

I have manually re-run them ... but to coverage is still broken. Let me force this merge. Sorry for the trouble.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

It looks like the project coverage is now green so you should be able to merge it.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub set status to `closed`|
|-|-|

In changeset bd34eb3cb09791f4c7eecb831a468743b2fe52b1
```
#!CommitTicketReference repository="" revision="bd34eb3cb09791f4c7eecb831a468743b2fe52b1"
Merge pull request #313 from twisted/8584-rodrigc-tcp

Author: rodrigc
Reviewer: adiroiban
Fixes: #8584

Increase test coverage of test_tcp_internals.py on Python 3
```
